topics/icrc_33_call_canister.md

```diff
@@ -0,0 +1,191 @@
+
+    RP ->> S: Request canister call
+    alt Version is not supported
+        S ->> RP: Error response: Version not supported (20101)
+    else Relying party has not been granted the `icrc33_call_canister` permission scope<br>or the request does not comply with scope restrictions
+        S ->> RP: Error response: Permission not granted (30101)
+    else
+        alt Canister supports ICRC-21
+            Note over S,C: Follow the ICRC-21 standard
+        else Canister does not support ICRC-21
+            S ->> U: Display warning and canister call details (canisterId, sender, method, arg)
+            Note over S,U: The warning should inform the user that the canister does not support ICRC-21<br/>The arguments should be decoded, otherwise another warning must be displayed
+        end
+        alt Approved
+            U ->> S: Approve request
+            S ->> C: Submit canister call
+            S ->> S: Wait for the canister call result
+            alt Call successful
+                S ->> U: Display success message
+                S ->> RP: Canister call response
+                RP ->> RP: Validate the certificate
+                RP ->> RP: Retrieve the result
+            else Call failed
+                S ->> U: Display failure message
+                S ->> RP: Error response: Network error (40001) | Unknown error (10001)
+            end
+        else Rejected
+            U ->> S: Reject request
+            S ->> RP: Error response: Action aborted (30201)
+        end
+    end
+```
+
+
+## Scope (according to the [ICRC-25 standard](./icrc_25_signer_interaction_standard.md))
+
+**Method:** `icrc33_call_canister` 
+
+**Optional Properties:**
+- `targets` (`text` array): A list of target canister ids (textual representation) the scope is restricted to. If the list is not present, the scope applies to all canisters (i.e. the permission is not restricted).
+- `senders` (`text` array): A list of sender principal ids (textual representation) the scope is restricted to. If the list is not present, the scope applies to all senders (i.e. the permission is not restricted).
+
+**Example:**
+```json
+{
+    "id": 1,
+    "jsonrpc": "2.0",
+    "method": "icrc25_request_permissions",
+    "params": {
+        "version": "1",
+        "scopes": [
+            {
+                "method": "icrc33_call_canister",
+                "targets": ["ryjl3-tyaaa-aaaaa-aaaba-cai"],
+                "senders": ["ryjl3-tyaaa-aaaaa-aaaba-cai"]
+            }
+        ]
+    }
+}
+```
+
+## Errors
+
+See [ICRC-25](./icrc_25_signer_interaction_standard.md#errors-3) for a list of errors that can be returned by all methods.
+
+While processing the request from the relying party, the signer can cancel it at any time by sending an [error](#errors) in response. In addition to the pre-defined JSON-RPC 2.0 errors ([-32600 to -32603 and -32700](https://www.jsonrpc.org/specification#error_object)), the following values are applicable:
+- `30201 Action aborted`
```

Comment:
Not necessarily in this PR, but we need to do another pass through all other methods as well to make sure that `insufficent_scope` and `denied_by_user` are well defined and clearly distinguishable. Maybe we should rename the errors in the base standard as well to more intuitively map to the cause of the error.